### ISSUE TITLE
Homebridge-sector breaks due to outdated version of sectoralarm-package

### ISSUE THREAD

**Issue by josteink** [frli4797/homebridge-sector#10]
Homebridge-sector breaks on start with an error-message [about an API error which seems to be resolved in latest sectoralarm-package](https://github.com/perbrage/sectoralarm/commit/6b1f7ab784a1987a19833ae4a11fc120f26a5f76):

````
[11/2/2019, 11:43:05 AM] [Home security] Cannot create property 'lastInteractionBy' on string 'INVALID_VERSION'
[11/2/2019, 11:43:05 AM] [Home security] undefined
(node:1763) UnhandledPromiseRejectionWarning: TypeError: Cannot create property 'lastInteractionBy' on string 'INVALID_VERSION'
    at /usr/lib/node_modules/homebridge-sector/node_modules/sectoralarm/lib/site.js:30:40
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
(node:1763) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1763) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
````

Using the [default boilerplate/example](https://github.com/perbrage/sectoralarm/blob/master/README.md#usage-example) for `sectoralarm` directly I get the expected output:

````
$ node index.js 
{ siteId: '234798432',
  name: 'Mitt alarmsystem',
  partialArmingAvailable: true,
  annexArmingAvailable: false,
  locksAvailable: false,
  smartPlugsAvailable: false,
  temperaturesAvailable: true,
  camerasAvailable: false,
  locks: [],
  temperatures: [ {}, {} ],
  smartPlugs: [],
  cameras: [] }
{"siteId":"234798432","name":"Mitt alarmsystem","armedStatus":"disarmed","annexArmedStatus":"unknown"}
{ siteId: '234798432',
  name: 'Mitt alarmsystem',
  armedStatus: 'disarmed',
  annexArmedStatus: 'unknown' }
[ { time: '2019-11-02 06:44:17', action: 'disarmed', user: 'M' },
  { time: '2019-11-01 22:04:41', action: 'partialArmed', user: '' },
  { time: '2019-10-30 06:49:49', action: 'disarmed', user: '3' },
  { time: '2019-10-29 21:40:31', action: 'partialArmed', user: '' },
  { time: '2019-10-29 16:21:21', action: 'disarmed', user: '3' },
  { time: '2019-10-29 07:19:48', action: 'armed', user: '' },
  { time: '2019-10-29 06:03:34', action: 'disarmed', user: 'M' },
  { time: '2019-10-28 22:17:42', action: 'partialArmed', user: '' },
  { time: '2019-10-28 16:19:13', action: 'disarmed', user: '3' },
  { time: '2019-10-28 07:11:48', action: 'armed', user: '' } ]
[ { sensorId: '2250103710E', name: 'stue', temperature: '23' },
  { sensorId: '2250103711D', name: 'kjeller ', temperature: '24' } ]
````

Basically, the underlying package works 100% for all my use-cases, which are more use-cases then the ones currently used by `homebridge-sector`.

This leads me to believe that simply upgrading the `sectoralarm`-package to the latest version (and fix up API-invocations where required) might be sufficient to fix this error.

**Comment by perbrage** [comment]
As the author of the underlying package I can confirm that this package simply needs to upgrade to the latest version.

Sector Alarm have for the past 6-8 months started to require a version tag for some of the api calls, even though there are no changes and/or older versions available.

I added a feature to the general library (the underlying one to this project) so it can auto-detect the version to use at startup. So after upgrading if there are version errors a simple restart would fix it. In the future I might add a self-resolve feature to avoid version errors completly, unless they actually do breaking changes.

Regards,
Per

**Comment by josteink** [comment]
Old issue is old. Not sure if it's still even an issue. Abandoning.
